integrationtest/integration_test.go

```diff
@@ -18,25 +20,188 @@ import (
 //   - A Kubernetes cluster in which:
 //       - it can use the `test` namespace
 //       - Vault is deployed and accessible
+//       - There is a serviceaccount called test-token-reviewer-account with access to the TokenReview API
 // See `make setup-integration-test` for manual testing.
 func TestMain(m *testing.M) {
 	if os.Getenv("INTEGRATION_TESTS") != "" {
-		cmd := exec.Command("kubectl", "exec", "--namespace=test", "vault-0", "--", "cat", "/var/run/secrets/kubernetes.io/serviceaccount/token")
-		out := &bytes.Buffer{}
-		cmd.Stdout = out
-		cmd.Stderr = out
-		if err := cmd.Run(); err != nil {
-			fmt.Println(out.String())
-			fmt.Println(err)
-			os.Exit(1)
-		}
 		os.Setenv("VAULT_ADDR", "http://127.0.0.1:38200")
 		os.Setenv("VAULT_TOKEN", "root")
-		os.Setenv("KUBERNETES_JWT", out.String())
+		os.Setenv("KUBERNETES_JWT", getKubernetesJwt())
+		os.Setenv("TOKEN_REVIEWER_JWT", getTokenReviewerJwt())
 		os.Exit(m.Run())
 	}
 }
 
+func getTokenReviewerJwt() string {
+	name := runCmd("kubectl --namespace=test get serviceaccount test-token-reviewer-account -o jsonpath={.secrets[0].name}")
+	b64token := runCmd(fmt.Sprintf("kubectl --namespace=test get secrets %s -o jsonpath={.data.token}", name))
+	token, err := base64.URLEncoding.DecodeString(b64token)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
```

Comment:
I could also switch to using a `panic`, so that the `defer` functions would be called as normal. I'll do that for now, as it is simpler, but let me know if you'd prefer the sub-test structure.

Reply:
Yeah that sounds like a good solution too